### DISCUSSION TITLE
Using Variant Groups & CSS Directives with Runtime JS

### DISCUSSION THREAD

**Discussion by Deanphillips** [unocss/unocss#906] [Q&A]
The runtime CDN is fantastic (really excited to use it with the page builder we're using)!

I don't see a way to use Variant Groups & CSS Directives with this? If it's possible, could someone please point me in the direction of getting it working, please? 🙏

**Comment by antfu** [comment]
They are compile-time transformers, meaning to change the code at the build time. I would avoid using them in the runtime (while it should be possible, feel free to experiment if you really want it)

**Reply by Deanphillips** [reply]
Thanks, @antfu! Good to know, how would we import them via the script?

**Reply by antfu** [reply]
In the runtime, you can't actually transform the code. All you can do is to get the attributes and replace them if they contain the transformation. Basically, you need to do:

```ts
// for every dom element
const el = document.getElement(/* ... */)
if (includesVariantGroup(el.class)) {
  el.class = transform(el.class)
}
```

Which makes it very inefficient, and introduces many issues to deal with (how to prevent infinite loops, timing to update, etc.)

Again, in general, I am won't recommend doing this.
